Session.js

```diff
@@ -113,6 +112,62 @@ module.exports = class Session {
     return allCandidates[0];
   }
 
+  /**
+   * conditionalDecide
+   * Decision options to determine decision to take.
+   *
+   * @param  {string} name - name of event
+   * @param  {array} candidates - variations to choose from
+   * @param  {array} contexts - contexts to choose from
+   * @param  {Object} options (optional) - timeout
+   * @param  {Function} cb - error and decision
+   */
+  conditionalDecide(name, candidates = [], event, contexts = [], options = {}, cb) {
+    if (!event || event === '') throw new Error('Event name required for conditional decide.');
+    if (!Array.isArray(contexts) || contexts.length === 0) throw new Error('Contexts required for conditional decide.');
+
+    options.timeout = options.timeout || this.timeout;
+    options.url = this.amp.domain + this.amp.apiPath + this.amp.key + "/conditionalDecide";
+
+    const { requestSafeCandidates, allCandidates } = this._formatCandidates(candidates);
+
+    if (utils.isFunction(arguments[arguments.length - 1])) cb = arguments[arguments.length - 1];
+    
+    options.limit = options.limit || allCandidates.length
+
+    if (allCandidates.length > 50) {
+      if (cb) {
+        cb(new Error("Candidate length must be less than or equal to 50."), allCandidates[0]);
+      }
+
+      return allCandidates[0];
+    }
+
+    this.request({
+      name: name,
+      event: event,
+      key: this.amp.key,
+      sessionId: this.id,
+      userId: this.userId,
+      contexts: contexts,
+      decision: {
+        candidates: requestSafeCandidates,
+        limit: options.limit
+      },
+      index: this.index++
+    }, options, (err, response, body) => {
+      if (err || (!body || !body.indexes)) {
+        // use default
+        if(cb) cb(err, allCandidates[0]);
+      } else {
+        const rankedCandidates = body.indexes.map(index => allCandidates[index]);
+        if (cb) cb(null, rankedCandidates, body);
+      }
+    });
+
+    return allCandidates[0];
```

Comment:
shouldn't the return above be a list?

Reply:
Yes.  Updated for error condition as well.